src/Model_Hydro/MHD_Init_BField_ByFile.cpp

```diff
@@ -0,0 +1,419 @@
+//
+// Note        :  
+//
+// Parameter   :  B_lv         : Target AMR level
+//
+// Return      :  amr->patch->magnetic
+//-------------------------------------------------------------------------------------------------------
+void MHD_Init_BField_ByFile( const int B_lv )
+{
+
+#  ifndef MHD
+   Aux_Error( ERROR_INFO, "MHD must be enabled !!\n" );
+#  endif
+
+#  ifndef SUPPORT_HDF5
+   Aux_Error( ERROR_INFO, "SUPPORT_HDF5 must be set to load a vector potential from a file !!\n" );
+#  endif
+
+   if ( MPI_Rank == 0 )    Aux_Message( stdout, "   Loading the magnetic field from the input file ...\n" );
+
+   const char B_Filename[] = "B_IC";
+
+   const double dh       = amr->dh[B_lv];
+
+   double *Axf, *Ayf, *Azf;
+
+   if ( !Aux_CheckFileExist(B_Filename) )
+      Aux_Error( ERROR_INFO, "file \"%s\" does not exist !!\n", B_Filename );
+
+// Open the magnetic field file and determine the dimensionality of the vector
+// potential grid
+
+#  ifdef SUPPORT_HDF5
+
+   herr_t status;
+   hid_t dataset, dataspace;
+   hsize_t dims[3], maxdims[3];
+
+   hid_t mag_file_id = H5Fopen(B_Filename, H5F_ACC_RDONLY, H5P_DEFAULT);
+
+   if ( B_lv == 0 ) {
+
+     dataset = H5Dopen(mag_file_id, "magnetic_vector_potential_z", H5P_DEFAULT);
+     
+     dataspace = H5Dget_space(dataset);
+   
+     H5Sget_simple_extent_dims(dataspace, dims, maxdims);
+
+     H5Sclose(dataspace);
+     H5Dclose(dataset);
+
+//   NOTE: Magnetic vector potential arrays are stored in column-major order,
+//   i.e., Ax[nAx][nAy][nAz]
+
+     nAx = dims[0];
+     nAy = dims[1];
+     nAz = dims[2];
+
+   }
+
+#  endif
+
+// Read the coordinate information from the vector potential grid
+   Axcoord = new double [ nAx ];
+   Aycoord = new double [ nAy ];
+   Azcoord = new double [ nAz ];
+
+#  ifdef SUPPORT_HDF5
+
+   dataset = H5Dopen(mag_file_id, "x", H5P_DEFAULT);
+   status  = H5Dread(dataset, H5T_NATIVE_DOUBLE, H5S_ALL,
+      H5S_ALL, H5P_DEFAULT, Axcoord);
+   H5Dclose(dataset);
+
+   dataset = H5Dopen(mag_file_id, "y", H5P_DEFAULT);
+   status  = H5Dread(dataset, H5T_NATIVE_DOUBLE, H5S_ALL,
+      H5S_ALL, H5P_DEFAULT, Aycoord);
+   H5Dclose(dataset);
+
+   dataset = H5Dopen(mag_file_id, "z", H5P_DEFAULT);
+   status  = H5Dread(dataset, H5T_NATIVE_DOUBLE, H5S_ALL,
+      H5S_ALL, H5P_DEFAULT, Azcoord);
+   H5Dclose(dataset);
+
+#  endif
+
+// Cell spacing and left edge of vector potential grid
+
+   Adx = Axcoord[1]-Axcoord[0];
+   Ady = Aycoord[1]-Aycoord[0];
+   Adz = Azcoord[1]-Azcoord[0];
+
+   Axmin = Axcoord[0]-0.5*Adx;
+   Aymin = Aycoord[0]-0.5*Ady;
+   Azmin = Azcoord[0]-0.5*Adz;
+
+   double Axmax = Axcoord[nAx-1]+0.5*Adx;
+   double Aymax = Aycoord[nAy-1]+0.5*Ady;
+   double Azmax = Azcoord[nAz-1]+0.5*Adz;
+
+   if ( amr->BoxEdgeL[0] < Axmin || amr->BoxEdgeR[0] > Axmax ||
+        amr->BoxEdgeL[1] < Aymin || amr->BoxEdgeR[1] > Aymax ||
+        amr->BoxEdgeL[2] < Azmin || amr->BoxEdgeR[2] > Azmax ) 
```

Comment:
Since now we need three buffer cells, it seems to me that the check should be
```
if ( amr->BoxEdgeL[0] < Axmin+2*Adx || amr->BoxEdgeR[0] >= Axmax-2*Adx ||
     amr->BoxEdgeL[1] < Aymin+2*Ady || amr->BoxEdgeR[1] >= Aymax-2*Ady ||
     amr->BoxEdgeL[2] < Azmin+2*Adz || amr->BoxEdgeR[2] >= Azmax-2*Adz )
```